### PR TITLE
Quick set shooter

### PR DESCRIPTION
**Pull request by HarrisHamid**
**Purpose**

The purpose of this PR was to fix the QuickSetShooter setRedZone not responding issue and adjusted piston setting numbers. Also commented out double solenoid code that was causing errors.

**Description**
As a group, we fixed:

 setRedZone
- corrected issue where isFinished was set to false
commented out double solenoids
changed the module number, forward channel, and reverse channel for the pistons

**Testing**
Currently, this code builds and shooting and firing at different zones have been tested

### CODE REVIEW

**Comment by bjmcternan** [comment]
@HarrisHamid This PR does a lot more than just fix the RED zone issue that we saw. This is a complete revamp of the firing mechanism and RPM control. There's also code for the piston manipulation.

Don't forget to mention how the XBox button mapping! Please edit the description

File: src/main/java/frc/robot/RobotContainer.java [hunk 1]
```diff
@@ -109,52 +97,80 @@ private void configureButtonBindings()
...
+        final POVButton dpadDownButton = new POVButton(xboxController, 180);
+        dpadDownButton.whenPressed(new ShooterPistonDown(m_launcher));
         // END AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=BUTTONS
         }
 
     // BEGIN AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=FUNCTIONS
     public Joystick getleftJoy() 
     {
         return leftJoy;
+
```
Comment (bjmcternan) [inline src/main/java/frc/robot/RobotContainer.java:173]: Extra whitespace, this is OK for now

File: src/main/java/frc/robot/commands/Fire.java [hunk 2]
```diff
@@ -51,18 +54,27 @@ public void initialize()
...
+        {
+            //Feed!
+            m_feeder.beltOn();
+            m_feeder.feedOn();
+        }
+        else
+        {
+            //Not ready! Don't Feed!
+            m_feeder.feedOff();
+    	    m_feeder.beltOff();
```
Comment (bjmcternan) [inline src/main/java/frc/robot/commands/Fire.java:68]: Tab issue, this is OK for now

File: src/main/java/frc/robot/commands/Fire.java [hunk 3]
```diff
@@ -51,18 +54,27 @@ public void initialize()
...
+        }
     }
 
     // Called once the command ends or is interrupted.
     @Override
     public void end(boolean interrupted) 
     {
-    	m_feeder.feedOff();
+        m_feeder.feedOff();
```
Comment (bjmcternan) [inline src/main/java/frc/robot/commands/Fire.java:76]: Tab issue, this is ok for now

File: src/main/java/frc/robot/commands/SetBlueZone.java [hunk 4]
```diff
@@ -78,4 +71,4 @@ public boolean runsWhenDisabled()
         return false;
         // END AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=DISABLED
     }
-}
+}
```
Comment (bjmcternan) [inline src/main/java/frc/robot/commands/SetBlueZone.java:74]: This needs a whitespace at the end of the file

File: src/main/java/frc/robot/commands/SetGreenZone.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    }
+
+    @Override
+    public boolean runsWhenDisabled() 
+    {
+        // BEGIN AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=DISABLED
+        return false;
+        // END AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=DISABLED
+    }
+}
```
Comment (bjmcternan) [inline src/main/java/frc/robot/commands/SetGreenZone.java:74]: This needs a white space at the end of the file

File: src/main/java/frc/robot/commands/SetRedZone.java [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+    // Returns true when the command should end.
+    @Override
+    public boolean isFinished() 
+    {
+        // BEGIN AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=DISABLED
+        return true;
+        // END AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=DISABLED
+    }
+}
```
Comment (bjmcternan) [inline src/main/java/frc/robot/commands/SetRedZone.java:68]: This needs a whitepace at the end of the file

File: src/main/java/frc/robot/commands/SetYellowZone.java [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+    }
+
+    // Returns true when the command should end.
+    @Override
+    public boolean isFinished() 
+    {
+        return true;
+        // END AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=DISABLED
+    }
+}
```
Comment (bjmcternan) [inline src/main/java/frc/robot/commands/SetYellowZone.java:67]: This needs a whitespace at the end of the file

File: src/main/java/frc/robot/subsystems/Launcher.java [hunk 8]
```diff
@@ -88,56 +76,76 @@ public void simulationPeriodic()
...
-        return conversionVel;
+
+    //Converts RPMs to encoder units per desiseconds (100ms)
+    private double convertRPMtoVelocity(double rpm) 
+    {   
+        return (rpm * GEAR_RATIO * ENCODER_UNITS_PER_REVOLUTION) / MINUTES_TO_DECISECONDS;
     }
 
-    public void enableLauncher(double rpm) 
+    //Converys Encoder units per desiseconds (100ms) to RPMs
```
Comment (bjmcternan) [inline src/main/java/frc/robot/subsystems/Launcher.java:86]: Misspelling, converts

File: src/main/java/frc/robot/subsystems/LauncherController.java [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+    {
+        // BEGIN AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=CONSTRUCTORS
+        // END AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=CONSTRUCTORS
+        m_launcherState = LauncherState.OFF;
+        m_launcher = launcher;
+    }
+
+
+    @Override
+    public void periodic() {
```
Comment (bjmcternan) [inline src/main/java/frc/robot/subsystems/LauncherController.java:53]: Fix the {}'s

File: src/main/java/frc/robot/subsystems/LauncherController.java [hunk 10]
```diff
@@ -0,0 +1,138 @@
...
+    *
+    */
+    public LauncherController(Launcher launcher) 
+    {
+        // BEGIN AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=CONSTRUCTORS
+        // END AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=CONSTRUCTORS
+        m_launcherState = LauncherState.OFF;
+        m_launcher = launcher;
+    }
+
```
Comment (bjmcternan) [inline src/main/java/frc/robot/subsystems/LauncherController.java:50]: Extra whitespace

File: src/main/java/frc/robot/subsystems/LauncherController.java [hunk 11]
```diff
@@ -0,0 +1,138 @@
...
+    @Override
+    public void periodic() {
+        // This method will be called once per scheduler run
+    }
+
+    @Override
+    public void simulationPeriodic() {
+        // This method will be called once per scheduler run when in simulation
+
```
Comment (bjmcternan) [inline src/main/java/frc/robot/subsystems/LauncherController.java:60]: Extra whitespace

**Review by bjmcternan — CHANGES_REQUESTED** [review]
There's some whitespace issues that needs to be addressed. While you're in there you might as well fix the TAB issues as well.

**Comment by HarrisHamid** [comment]
**Purpose**

The purpose of this PR was to create QuickSetShooter buttons on an Xbox controller. When a button is pressed, the motor connected to the shooter wheel will set the  RPM to the RPM mapped to that button.  Depending on the button pushed, either the speed would change or turn the motor off.  A fire button was also included to fire balls when held, which would also manipulate the launcher and feeder. Lastly, shooterpiston buttons were added in which the d-pad on an Xbox controller would allow it to extend or retract.

**Description**

As a group, we:

setShootingZones
- The XYBA button on an Xbox controller would have a certain RPM mapped depending on which zone they were assigned. A click of a button once would set the RPM to the specific speed, the same button again would turn the motor off, and a different button clicked would change the overall speed.
- An issue resolved was fixing the RedZone, which was not responding/ setting to the mapped RPM, by setting isFisnihed to true and correctly assigning the buttons in robotcontainer.
Fire Button
- The fire button was mapped to a bumper and would need to be held in order to fire since it controlled the feeder and launcher. Once it was let go, the feeder and launcher would disable, but the motor controlling the wheel would continue to spin.
- corrected issue where isFinished was set to false
ShooterPiston
- Controls the piston meant to manipulate the launcher angle for the green zone. The d-pad is used; when pushed up, it should extend the piston, while pushing it down should retract it.

-commented out double solenoids
-changed the module number, forward channel, and reverse channel for the pistons

**Testing**
Currently, this code builds, and shooting and firing at different zones has been tested. 
The shooter piston however is not operational with the d-pad.

**Review by bjmcternan — APPROVED** [review]
Looks good. There are some minor issues but we can get them on the next PR.

**Pull request merged**